### DISCUSSION TITLE
ファイルの一括アップロードと一括ダウンロード機能

### DISCUSSION THREAD

**Discussion by ekzemplaro** [increments/qiita-discussions#20] [要望・アイデア・不具合]
プログラムをコピーペーストでアップロードするのは、タイプミスの発生のもとです。zip に固めたファイルを一括でアップロードし、仮に付けられた名前を編集する機能が欲しい。
そして、ファイルを一括でダウンロードする機能も欲しい。

**Comment by kyntk** [comment]
@ekzemplaro ご意見ありがとうございます。
実現したいことについて、少し読みとれていないところがあるのですが、zipにファイルをアップロードして、そのファイル名を記事のマークダウンに埋め込むと、コードが展開されて表示できるというようなものでしょうか?

**Comment by ekzemplaro** [comment]
そこまでの変更ではなく、現状のコピーペースト作業をしないで済むようにするという機能です。画像については、アップロードができますが、プログラムはアップロードが出来ません。
例えば、次のようなケース
amain.py
funciton01.py
function02.py
function03.py
function04.py
という５つのファイルをアップロードする時、今は、コピーペーストの作業が５回必要です。
それを、zip で固めて送ると。
py:amain.py
py:funciton01.py
py:funciton02.py
py:funciton03.py
py:funciton04.py
と、展開されて欲しいものです。
ただ、ファイルの順番をどうするかに工夫が必要です。ソート順が良いと思います。
上の例では、amain.py が上に来るようにファイル名を変更してみました。

**Comment by kyntk** [comment]
@ekzemplaro なるほど、ファイルのドラッグアンドドロップでのアップロードと、zipを用いた複数ファイルのアップロードですね。
確かに、ファイル数が増えるとコピペが手間ですよね。

書く体験の改善するプロジェクトを予定していますので、そこの中で優先度議論させていただきます。
